main.go

```diff
@@ -25,6 +25,7 @@ import (
 	tr "github.com/open-cluster-management/search-collector/pkg/transforms"
 
 	"github.com/golang/glog"
+	"github.com/open-cluster-management/addon-framework/pkg/lease"
```

Comment:
Should this be in this PR?

Reply:
This PR is where I'm working on adding the lease. As I began working on the code, I had to setup the fork/merge workflow so I decided to squeeze that documentation into the same PR. This was actually the main thrust of this PR

Reply:
It sounds different from the PR's title!
In this case it's better to divide it so we can merge the README update and continue with the other work independently.

Reply:
Agreed. Let's use this PR that sherin has created, and keep this current PR small with only the doc updates. 

(I said to myself "I'll change the title to reflect the work once the code reflects the title", but now I'm realizing this PR can be more useful now if merged with doc updates only. No need to wait :) )

Let me remove the package import line.